alibi_detect/cd/metrics.py

```diff
@@ -0,0 +1,333 @@
+    p_vals = np.array(p_vals_list).flatten()
+    # TODO - is above OK? (to flatten over features for univariate detectors)
+
+    # QQ-plot
+    if qq_plot:
+        plot_qq(p_vals)
+    # Histogram plot
+    if hist_plot:
+        plot_hist(p_vals)
+    # Show plots
+    if qq_plot or hist_plot:
+        plt.show()
+
+    return p_vals
+
+
+def eval_test_power(
+    detector: type,  # TODO: type w/ BaseDriftDetector if introduced.
+    X_ref: Union[np.ndarray, list],
+    X_drift: Union[np.ndarray, list],
+    C_ref: Optional[np.ndarray] = None,
+    C_drift: Optional[np.ndarray] = None,
+    model: Optional[Callable] = None,
+    detector_kwargs: Optional[dict] = None,
+    sig_levels: np.ndarray = np.linspace(0.05, 0.5, 10),
+    n_runs: int = 100,
+    n_samples: Tuple[int, int] = (500, 500),
+    return_auc: bool = True,
+    power_plot: bool = False,
+    random_seed: Optional[Union[int, np.random.Generator]] = None
+) -> Union[np.ndarray, Tuple[np.ndarray, np.ndarray]]:
+    """
+
+    Parameters
+    ----------
+
+    Returns
+    -------
+
+    """
+    if (C_ref is None) != (C_drift is None):
+        raise ValueError("`C_ref` and `C_drift` must both be `None`, or both be given as np.ndarray's.")
+
+    # Check data sizes
+    n_ref, n_drift = len(X_ref), len(X_drift)
+    _check_sufficient_data_size(X_ref, n_samples, n_runs, X_test=X_drift)
+
+    # Setup detector_kwargs
+    if detector_kwargs is None:
+        detector_kwargs = {}
+    detector_kwargs.pop('p_val', None)  # Need to set this ourselves
+
+    # Perform preprocessing (if preprocess_fn exists) to save repeated compute in loop
+    preprocess_fn = detector_kwargs.pop('preprocess_fn', None)
+    if preprocess_fn is not None:
+        X_ref = preprocess_fn(X_ref)
+        X_drift = preprocess_fn(X_drift)
+
+    # NumPy RNG
+    rng = np.random.default_rng(random_seed)
+
+    # Main experiment loop
+    power = np.zeros(len(sig_levels), dtype=float)
+    for _ in tqdm(range(n_runs)):
+        # Subsample data
+        idx_ref = rng.choice(n_ref, size=n_samples[0], replace=False)
+        idx_drift = rng.choice(n_drift, size=n_samples[1], replace=False)
+        x_ref, x_drift = X_ref[idx_ref], X_drift[idx_drift]
+        detector_args, predict_args = [x_ref], [x_drift]
+
+        if C_ref is not None:
+            c_ref, c_drift = C_ref[idx_ref], C_drift[idx_drift]
+            detector_args.append(c_ref)
+            predict_args.append(c_drift)
+
+        if model is not None:
+            detector_args.append(model)
+
+        # Init detector and predict
+        for i, sig in enumerate(sig_levels):
+            dd = detector(*detector_args, p_val=sig, **detector_kwargs)
+            preds = dd.predict(*predict_args)
+            power[i] += preds['data']['is_drift']
+    power /= n_runs
+
+    if power_plot:
+        plot_power(sig_levels, power)
+        plt.show()
+
+    if return_auc:
+        auc = compute_auc(sig_levels, power)
```

Comment:
Whilst this is, of course, *an* area under a curve, this is not AUC in the colloquial power-measuring sense (which is the area under a plot of FPRs vs TPRs at different significance levels). We should therefore use a different name or compute the AUC. I actually think both can be useful, although computing the true AUC of course additionally requires obtaining false positive rates.

Reply:
Good spot, thanks. I've renamed `return_auc` in `eval_test_power` to `return_power_auc`. We could even rename further if you think that's still confusing?

`eval_roc` now returns the true AUC.